contracts/swap/swap.go

```diff
@@ -45,14 +44,18 @@ type Backend interface {
 
 // Contract interface defines the methods exported from the underlying go-bindings for the smart contract
 type Contract interface {
-	// Withdraw attempts to withdraw Wei from the chequebook
-	Withdraw(auth *bind.TransactOpts, backend Backend, amount *big.Int) (*types.Receipt, error)
+	// Withdraw attempts to withdraw ERC20-token from the chequebook
+	Withdraw(auth *bind.TransactOpts, amount *big.Int) (*types.Receipt, error)
```

Comment:
It is not necessarily part of this PR, but just a cleanup (probably some legacy code here). The `backend` is already part of `simpleContract`, so passing it as a function parameter does not make a lot of sense. 
What is more, it could lead to inconsistencies in the code, so I think it is much better to keep it out and instead always use the backend with which we created the `simpleContract` instance.

Reply:
thanks for the clarification, sounds good :+1: